st2client/tests/unit/test_auth.py

```diff
@@ -154,6 +157,14 @@ def runTest(self, mock_gp):
 
         self.shell.run(args)
 
+        # TODO(mierdin): This tests that this particular command sends X-Auth-Token but you should
+        # also test other commands after this token has been installed
+        kwargs = {
+            'headers': {'X-Auth-Token': self.TOKEN['token'], 'content-type': 'application/json'},
```

Comment:
This test was correct since this request is hitting auth api (and not regular API) to authenticate to retrieve the token and as such, only username and password should be sent (and token returned in the response and used on subsequent requests to the st2 api).